### ISSUE TITLE
Global Search?

### ISSUE THREAD

**Issue by hoffi** [jansmolders86/mediacenterjs#183]
I always like to have an "app-wide" search, which searches just everywhere it can and displays the results categorized by the containing app for example. 

Not very easy to implement but what do you think?

**Comment by johnpool3** [comment]
This sounds like a great idea, a bit like spotlight on a mac perhaps.

**Comment by Jon889** [comment]
I started implementing this a few days ago, and it works, just knowing how to display the results is a bit ugly, I'm passing the name of the directive (view) in the list of apps (model).

**Comment by jansmolders86** [comment]
This is awesome guys! I have a couple of thoughts on the visual implementation. Do you mind me weighing in on this as well? I don't want to step on your feature if you just want to complete it yourself!

**Comment by Jon889** [comment]
Definitely! I don't know about the others, but I guess it's the same, but I don't consider anything "my" feature :)

**Comment by jansmolders86** [comment]
Excellent. I just don't want to step on anyone's toes here. I fully appreciate all the hard work you guys are doing!

**Comment by hoffi** [comment]
Hey @Jon889 
really like it already :+1: 

But i would find it more useful if the search is directly on the "dashboard".

**Comment by Jon889** [comment]
I quite agree :) though it seemed like that would be easier to do after making it a single page app (which is necessary to do things like background audio, or shrinking the video player picture in picture style)

**Comment by jansmolders86** [comment]
Sorry I've taken so long to respond guys! I've been work pretty much day and night for the last couple of weeks. It'll soon get less busy and when it does I want to prioritize making MCJS a complete onepage angular app which ties in nicely with this issue as Jon mentioned!

**Comment by Jon889** [comment]
Work should definitely be the priority, though hopefully you don't work yourself to death :) I'm curious what you ideas were for the search?

**Comment by jansmolders86** [comment]
Thanks for understanding. I just don't want you guys to get the feeling I'm abandoning this project. :)

I want to express again I totally love this feature! It's really well done already! And super fast!  
I just have a ton of ideas to use this once we have the entire thing in Angular.

I would love to see a Windows 8+ esk search capability that would really fit the project! 
I'll open an issue for the anugalrisation of the entire projects to discuss it's features.

**Comment by jansmolders86** [comment]
So, shall we merge this in it's current state? I love the current functionality already and I hate to see us waist this work until we move to the 2.0 version.

**Comment by Jon889** [comment]
There are some parts of the implementation I'm not quite happy with, there isn;t really a way to search multiple types of things from the same app (for example tracks and albums in the music app). And the way it knows what directive to use to display the results is odd/too connected.
